pkg/router/template/router.go

```diff
@@ -953,22 +967,66 @@ func generateDestCertKey(config *ServiceAliasConfig) string {
 }
 
 // getServiceUnits returns a map of service keys to their weights.
-// Weight suggests the % of traffic that a given service will receive
-// compared to other services pointed to by the route.
-func getServiceUnits(route *routeapi.Route) map[string]int32 {
+// The requests are loadbalanced among the services referenced by the route.
+// The weight (0-256, default 1) sets the relative proportions each
+// service gets (weight/sum_of_weights fraction of the requests).
+// When the weight is 0 no traffic goes to the service. If they are
+// all 0 the request is returned with 503 response.
+// For each service, the requests are distributed among the endpoints.
+// Each endpoint gets weight/numberOfEndpoints portion of the requests.
+// The above assumes roundRobin scheduling.
+func getServiceUnits(r *templateRouter, route *routeapi.Route) map[string]int32 {
 	serviceUnits := make(map[string]int32)
```

Comment:
Please just get the service units using the call, and then adjust the weights in the structure after first making a pass to determine the max.